### PR TITLE
Upgrade ncurses to 6.3 [patch 20220612] to fix CVE-2022-29458

### PR DESCRIPTION
**Pull request by jslobodzian**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/tools/cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 

What does the PR accomplish, why was it needed?

###### Change Log  

- Change
- Change
- Change

###### Does this affect the toolchain?  

**YES/NO**

###### Associated issues  

- #xxxx

###### Links to CVEs  
- https://nvd.nist.gov/vuln/detail/CVE-YYYY-XXXX

###### Test Methodology

- Pipeline build id: xxxx

### CODE REVIEW

File: cgmanifest.json [hunk 1]
```diff
@@ -4426,7 +4426,7 @@
         "other": {
           "name": "ncurses",
           "version": "6.3",
-          "downloadUrl": "https://invisible-mirror.net/archives/ncurses/ncurses-6.3.tar.gz"
+          "downloadUrl": "https://invisible-mirror.net/archives/ncurses/current/ncurses-6.3-20220612.tar.gz"
```
Comment (oliviacrain) [inline cgmanifest.json:4429]: The `current` in this URL makes me think it'll break when 6.4 is released- is there a more permanent URL we can use here?

File: toolkit/scripts/toolchain/container/toolchain-remote-wget-list [hunk 2]
```diff
@@ -17,7 +17,7 @@ http://ftp.gnu.org/gnu/make/make-4.2.1.tar.gz
 https://www.kernel.org/pub/linux/docs/man-pages/Archive/man-pages-5.02.tar.xz
 https://ftp.gnu.org/gnu/mpc/mpc-1.1.0.tar.gz
 http://www.mpfr.org/mpfr-4.0.1/mpfr-4.0.1.tar.xz
-https://invisible-mirror.net/archives/ncurses/ncurses-6.2.tar.gz
+https://invisible-mirror.net/archives/ncurses/current/ncurses-6.3-20220612.tgz
```
Comment (oliviacrain) [inline toolkit/scripts/toolchain/container/toolchain-remote-wget-list:20]: Same comment as the URL in the cgmanifest

**Review by oliviacrain — APPROVED** [review]
LGTM- just one nit regarding the source URL

**Review by anphel31 — APPROVED** [review]

**Comment by anphel31** [comment]
Please also update: toolkit/scripts/toolchain/container/toolchain-sha256sums

**Review by anphel31 — CHANGES_REQUESTED** [review]
please update new source in toolkit/scripts/toolchain/container/toolchain-sha256sums

File: cgmanifest.json (hunk 1 above)
Comment (anphel31) [reply cgmanifest.json:4429]: alternately, we could continue to use the original 6.3 source and check in the patch. (http://invisible-mirror.net/archives/ncurses/6.3/ncurses-6.3-20220612.patch.gz)

Comment (anphel31) [reply cgmanifest.json:4429]: FWIW, I wasn't able to find a more permanent URL for the patched sources

Comment (jslobodzian) [reply cgmanifest.json:4429]: I was reluctant to take just the patch....preferred to take the accumulated.  
And...yes, I think a better link is more desirable, but letting it go as is.

File: toolkit/scripts/toolchain/container/toolchain-remote-wget-list (hunk 2 above)
Comment (jslobodzian) [reply toolkit/scripts/toolchain/container/toolchain-remote-wget-list:20]: same response as for the URL

**Review by anphel31 — APPROVED** [review]

File: cgmanifest.json (hunk 1 above)
Comment (oliviacrain) [reply cgmanifest.json:4429]: Sounds good- I agree that using patched sources is cleaner despite the issues.

**Pull request merged**